src/test/java/com/puppycrawl/tools/checkstyle/AbstractModuleTestSupport.java

```diff
@@ -48,6 +48,22 @@
 
 public abstract class AbstractModuleTestSupport extends AbstractPathTestSupport {
 
+    /**
+     * Enum to specify options for checker creation.
+     */
+    public enum CheckerOptions {
+        /**
+         * Points that the module configurations
+         * has to be added under {@link TreeWalker}.
+         */
+        WITH_TREEWALKER,
+        /**
+         * Points that checker will be created as
+         * a root of default configuration.
+         */
+        DEFAULT
```

Comment:
DEFAULT is weird and most unclear state.

first of all we need to change name of option to ModuleCreationOption {IN_CHECKER, IN_TREEWALKER}